packages/shared/refThrottled/index.ts

```diff
@@ -10,18 +12,23 @@ import { useThrottleFn } from '../useThrottleFn'
  * @param  delay  A zero-or-greater delay in milliseconds. For event callbacks, values around 100 or 250 (or even higher) are most useful.
  * @param [trailing] if true, update the value again after the delay time is up
  * @param [leading] if true, update the value on the leading edge of the ms timeout
+ * @param [defaultClone] default clone function for non-primitive values
  */
-export function refThrottled<T>(value: Ref<T>, delay = 200, trailing = true, leading = true) {
+export function refThrottled<T>(value: Ref<T>, delay = 200, trailing = true, leading = true, defaultClone = _defaultClone) {
   if (delay <= 0)
     return value
 
-  const throttled: Ref<T> = ref(value.value as T) as Ref<T>
+  const _isObject = isObject(value.value)
+
+  const throttled: Ref<T> = ref(_isObject ? defaultClone(value.value) : value.value) as Ref<T>
 
   const updater = useThrottleFn(() => {
-    throttled.value = value.value
+    throttled.value = _isObject ? defaultClone(value.value) : value.value
```

Comment:
You should wrap a try-catch block to prevent error from throwing out

Reply:
<img width="800" alt="image" src="https://github.com/vueuse/vueuse/assets/58617090/f4c515e3-3536-4179-a9a1-a357be0b135f">
Like this?

Reply:
Hmm, I think we should check if the `cloneHandler` is our default handler or the user's custom handler. If it's the default handler and it throw an error, we can log an error message and assign `toValue(value)` to `throttled.value`. But if it's user's custom handler, maybe we just throw the error out.